MinecraftClient/Protocol/ProtocolHandler.cs

```diff
@@ -432,15 +442,102 @@ private static LoginResult MojangLogin(string user, string pass, out SessionToke
             }
         }
 
-        private static LoginResult MicrosoftLogin(string email, string password, out SessionToken session)
+        /// <summary>
+        /// Sign-in to Microsoft Account without using browser. Only works if 2FA is disabled.
+        /// Might not work well in some rare cases.
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="password"></param>
+        /// <param name="session"></param>
+        /// <returns></returns>
+        private static LoginResult MicrosoftMCCLogin(string email, string password, out SessionToken session)
+        {
+            var ms = new XboxLive();
+            try
+            {
+                var msaResponse = ms.UserLogin(email, password, ms.PreAuth());
+                return MicrosoftLogin(msaResponse, out session);
+            }
+            catch (Exception e)
+            {
+                session = new SessionToken() { ClientID = Guid.NewGuid().ToString().Replace("-", "") };
+                ConsoleIO.WriteLineFormatted("§cMicrosoft authenticate failed: " + e.Message);
+                if (Settings.DebugMessages)
+                {
+                    ConsoleIO.WriteLineFormatted("§c" + e.StackTrace);
+                }
+                return LoginResult.WrongPassword; // Might not always be wrong password
+            }
+        }
+
+        /// <summary>
+        /// Sign-in to Microsoft Account by asking user to open sign-in page using browser. 
+        /// </summary>
+        /// <remarks>
+        /// The downside is this require user to copy and paste lengthy content from and to console.
+        /// Sign-in page: 218 chars
+        /// Response URL: around 1500 chars
+        /// </remarks>
+        /// <param name="session"></param>
+        /// <returns></returns>
+        public static LoginResult MicrosoftBrowserLogin(out SessionToken session)
+        {
+            var ms = new XboxLive();
+            string[] askOpenLike =
+            {
+                "Copy the following link to your browser and login to your Microsoft Account",
+                ">>>>>>>>>>>>>>>>>>>>>>",
+                "",
+                ms.SignInUrl,
+                "",
+                "<<<<<<<<<<<<<<<<<<<<<<",
+                "When you see a blank page after signing in, copy the link from browser and paste it below"
+            };
+            ConsoleIO.WriteLine(string.Join("\n", askOpenLike));
+            string[] parts = { };
+            while (true)

```

Comment:
Would be nice to have a way to cancel, e.g. pressing Enter without typing anything.